queries.js

```diff
@@ -0,0 +1,96 @@
+const promise = require('bluebird')
+
+const options = {
+  promiseLib: promise
+}
+
+const pgp = require('pg-promise')(options)
+const connectionString = 'postgres://localhost:5432/possiblegazerbookstore'
+const db = pgp(connectionString)
+
+function getAllBooks(request, response, next) {
```

Comment:
I've seen this pattern a lot recently.  It's not *wrong* but I just wanted to share my opinion on why I wouldn't do it.  For me, the purpose of a module is to isolate responsibilities in my system.  In order to create code that is as isolated as possible, I try to minimize the dependencies between this module and other modules or frameworks that are external to the responsibility.

In the database module, I would create logic responsible only for connecting to the database and retrieving information.  This logic does that, but is *tightly coupled* with the express framework - retrieving the information in a middleware function.  I would prefer that my database functions could operate independently of the express framework, being responsible only for data retrieval.  The controller logic (the routes) themselves should have the responsibility of using these encapsulated pieces of database functionality in order to generate the response to the client.

As such, I would consider refactoring the database functions to be independent of the web framework, and return a promise to retrieve the requested data.  The promise would then be used in the controller logic to respond to the client:
#### database/queries.js
```js
const getAllBooks = () => db.any( 'SELECT * FROM books' )
```
#### routes/index.js
```js
router.get( '/', (request, response, next) => {
  db.getAllBooks()
    .then( data => response.status( 200 ).json({ data }) )
})
```